platform/include/platform/mbed_retarget.h

```diff
@@ -468,9 +468,16 @@ typedef struct DIR_impl DIR;
 #ifndef  ELIBEXEC
 #define ELIBEXEC        83      /* Cannot exec a shared library directly */
 #endif
+
+#if defined ( __ICCARM__ )
+#undef  EILSEQ                  /* defined in IAR arm/inc/c/errno.h:  #define EILSEQ  36 */
+#define EILSEQ          84      /* Illegal byte sequence */
```

Comment:
I don't think we want to add this. From the discussion, I was questioning our earlier approach with undef like here (see https://github.com/ARMmbed/mbed-os/pull/14751).

The goal was to provide default values only if they  are not defined. if IAR defines it, we should leave it as it is.

Reply:
But this can't compile as value 36 is already used....

Reply:
It is failing at https://github.com/ARMmbed/mbed-os/blob/master/platform/source/mbed_retarget.cpp#L2102
as there are 2 cases with the same value

Reply:
Good test ! 

@kjbracey-arm what do to in this case ? fix it as it is here ?

Reply:
or @pan- but I think both are OoO these days.

Lets get back to this one the next week.